examples/command/portals/kafka/instaclustr/docker/cluster_manager.sh

```diff
@@ -0,0 +1,160 @@
+}
+
+# Function to retrieve cluster details
+get_cluster_details() {
+  local cluster_id=$1
+  local response
+  response=$(curl -s -X GET -H "Content-Type: application/json" -u "${INSTACLUSTR_USER_NAME}:${INSTACLUSTR_API_KEY}" \
+    "https://api.instaclustr.com/cluster-management/v2/resources/applications/kafka/clusters/v2/${cluster_id}")
+  echo "${response}"
+}
+
+# Function to add a firewall rule
+add_firewall_rule() {
+    local cluster_id=$1
+    local my_ip=$(curl -s https://checkip.amazonaws.com)
+    local response
+
+    response=$(curl -s -X POST -H "Content-Type: application/json" -u "${INSTACLUSTR_USER_NAME}:${INSTACLUSTR_API_KEY}" \
+      "https://api.instaclustr.com/cluster-management/v2/resources/network-firewall-rules/v2" \
+      -d '{
+          "clusterId": "'"${cluster_id}"'",
+          "network": "'"${my_ip}/32"'",
+          "type":"KAFKA"
+        }')
+    #echo "${response}"
+    local status=$(echo "${response}" | jq -r '.status')
+    if [[ "${status}" == "GENESIS" ]]; then
+        echo "Firewall rule applied successfully"
+    else
+        echo "Firewall rule application failed"
+    fi
+}
+
+# Function to create a user for the Kafka cluster
+create_user() {
+  local cluster_id=$1
+  local response
+  response=$(curl -s -X POST -u "${INSTACLUSTR_USER_NAME}:${INSTACLUSTR_API_KEY}" \
+    "https://api.instaclustr.com/cluster-management/v2/resources/applications/kafka/users/v2" \
+    -H 'Content-Type: application/json' \
+    -d '{
+      "clusterId": "'"${cluster_id}"'",
+      "initialPermissions": "standard",
+      "options": {
+        "overrideExistingUser": false,
+        "saslScramMechanism": "SCRAM-SHA-256"
+      },
+      "password": "myPassword1.",
+      "username": "myKafkaUser"
+    }')
+  # echo "${response}"
+  if [[ $(echo "${response}" | jq -r '.id') != "null" && $(echo "${response}" | jq -r '.username') == "myKafkaUser" ]]; then
+      echo "User created successfully"
+  else
+      echo "User creation failed"
+  fi
+}
+
+# Function to delete the Kafka cluster
+delete_cluster() {
+  local cluster_id=$1
+  local response
+  echo "Deleting Kafka cluster"
+  response=$(curl -s -X DELETE -u "${INSTACLUSTR_USER_NAME}:${INSTACLUSTR_API_KEY}" \
+    "https://api.instaclustr.com/cluster-management/v2/resources/applications/kafka/clusters/v2/${cluster_id}")
+  echo "${response}"
+}
+
+# Main function
+main() {
+  if [[ "$1" == "cleanup" ]]; then
+    local cluster_id
+    cluster_id=$(cat cluster_id.txt)
+    delete_cluster "${cluster_id}"
+    rm cluster_id.txt
+    unset INSTACLUSTR_USER_NAME
+    unset INSTACLUSTR_API_KEY
+  else
+    local response=$(create_cluster)
+    local cluster_id=$(echo "${response}" | jq -r '.id')
+    echo "Creating a trial cluster. Cluster setup usually take around 5-10 minutes"
+    echo "Cluster ID: ${cluster_id}"
+    echo "${cluster_id}" > cluster_id.txt
+    wait_for_cluster "${cluster_id}"
+    create_user "${cluster_id}"
+    add_firewall_rule "${cluster_id}"
```

Comment:
I accidentally used a wrong API key and this went through without failing  with the following output:

```bash
Cluster ID: 
Current status: . Waiting for cluster to be running...
```

Perhaps we could check if the cluster id is empty before continuing.

Reply:
Good catch. Im capturing the response_code along with the response now. In such cases, the API response is empty. So had to do more parsing to get the response code separately. Could you give it a test again please?